bugbear.py

```diff
@@ -229,3 +254,13 @@ def compose_call_path(self, node):
             "to the exception.",
     type=BugBearChecker,
 )
+
+B901 = partial(
+    error,
+    message=("B901: Using ``yield`` together with ``return x``. Use ``raise "
+             "StopIteration(x)`` instead of ``return x`` if this was "
+             "intentional."),
```

Comment:
I was assuming that `raise StopIteration(x)` is a viable alternative to `return x`, so I guess recommending `# noqa` is the only option.